source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/layout/scrollingpage/ScrollingPageControlView.kt

```diff
@@ -0,0 +1,416 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package io.adaptivecards.renderer.layout.scrollingpage
+
+import android.animation.ArgbEvaluator
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.util.AttributeSet
+import android.util.SparseArray
+import android.view.View
+import androidx.annotation.ColorInt
+import androidx.viewpager.widget.ViewPager
+import androidx.viewpager2.widget.ViewPager2
+import io.adaptivecards.R
+import kotlin.math.abs
+
+/**
+ * A page indicator that displays dots for each page.
+ * Currently supports [ViewPager] - other sources can be added as needed.
+ * Adapted from: https://github.com/tinkoff-mobile-tech/ScrollingPagerIndicator/
+ */
+class ScrollingPageControlView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0,
+    configuration: ScrollingPageControlViewConfiguration ? = null
+) : View(context, attrs, defStyleAttr) {
+
+    /**
+     * Maximum number of dots which will be visible at the same time.
+     * If pager has more pages than visible_dot_count, indicator will scroll to show extra dots.
+     * Must be an odd number.
+     */
+    private var visibleDotCount = configuration?.visibleDotCount ?: resources.getInteger(R.integer.scrollingpagecontrolview_visibleDotCount)
+        set(value) {
+            require(value % 2 != 0) { "visibleDotCount must be odd" }
```

Comment:
"require(value: Boolean)" will throw an exception here if the value is even. Shouldn't this be caught?

Reply:
Have reference the same component from StarDust, I feel we should keep as is